### PR TITLE
imagej: 150 -> 153

### PR DESCRIPTION
**Pull request by doronbehar**
- Use 1 line per input argument.
- Don't use let ... in if not needed.
- Use ${version} in url string.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/imagej/default.nix [hunk 1]
```diff
@@ -1,48 +1,65 @@
...
+    # Read permisssion suffices for the jar and others.
+    # Simple cp shall clear suid bits, if any.
+    cp ij.jar $out/share/java
+    cp -dR luts macros plugins $out/share
+    mkdir $out/bin
+    makeWrapper ${jre}/bin/java $out/bin/imagej \
+      --add-flags "-jar $out/share/java/ij.jar -ijpath $out/share"
+  '' + lib.optionalString stdenv.isLinux ''
+    install -Dm644 ${icon} $out/share/icons/hicolor/128x128/apps/imagej.png
+    install -Dm644 ${desktopItem}/share/applications/ImageJ.desktop $out/share/applications/ImageJ.desktop
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/imagej/default.nix:49]: Why not use copyDesktopItems?

File: pkgs/applications/graphics/imagej/default.nix [hunk 2]
```diff
@@ -1,48 +1,65 @@
...
-        ImageJ is a public domain Java image processing program
-        inspired by NIH Image for the Macintosh.
-        It runs on any computer with a Java 1.4 or later virtual machine.
-      '';
-      license = licenses.publicDomain;
-      platforms = with platforms; linux ++ darwin;
-      maintainers = with maintainers; [ yuriaisaka ];
-    };
+    license = licenses.publicDomain;
+    platforms = with platforms; linux ++ darwin;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/imagej/default.nix:62]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/applications/graphics/imagej/default.nix [hunk 3]
```diff
@@ -1,48 +1,65 @@
...
+  # JAR files that are intended to be used by other packages
+  # should go to $out/share/java.
+  # (Some uses ij.jar as a library not as a standalone program.)
+  installPhase = ''
+    mkdir -p $out/share/java
+    # Read permisssion suffices for the jar and others.
+    # Simple cp shall clear suid bits, if any.
+    cp ij.jar $out/share/java
+    cp -dR luts macros plugins $out/share
+    mkdir $out/bin
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/imagej/default.nix:44]:
```suggestion

```

File: pkgs/applications/graphics/imagej/default.nix [hunk 4]
```diff
@@ -1,48 +1,65 @@
...
-      cp ij.jar $out/share/java
-      cp -dR luts macros plugins $out/share
-      mkdir $out/bin
-      makeWrapper ${jre}/bin/java $out/bin/imagej \
-        --add-flags "-jar $out/share/java/ij.jar -ijpath $out/share"
+  # JAR files that are intended to be used by other packages
+  # should go to $out/share/java.
+  # (Some uses ij.jar as a library not as a standalone program.)
+  installPhase = ''
+    mkdir -p $out/share/java
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/imagej/default.nix:39]:
```suggestion
    mkdir -p $out/share/java $out/bin
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 128077` at db21d393 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>imagej</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/graphics/imagej/default.nix:38:3:
```
   |
38 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/graphics/imagej/default.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/applications/graphics/imagej/default.nix:49]: Had no idea it exists! Thanks.

File: pkgs/applications/graphics/imagej/default.nix [hunk 5]
```diff
@@ -1,48 +1,68 @@
...
-  imagej150 = stdenv.mkDerivation {
-    pname = "imagej";
-    version = "150";
+  desktopItem = makeDesktopItem {
+    name = "ImageJ";
+    desktopName = "ImageJ";
+    icon = "imagej";
+    categories = "Science;Utility;Graphics;";
+    exec = "imagej";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/imagej/default.nix:18]:
```suggestion
```

File: pkgs/applications/graphics/imagej/default.nix [hunk 6]
```diff
@@ -1,48 +1,68 @@
...
-      sha256 = "97aba6fc5eb908f5160243aebcdc4965726693cb1353d9c0d71b8f5dd832cb7b";
-    };
-    nativeBuildInputs = [ makeWrapper unzip ];
+  src = fetchurl {
+    url = "https://wsr.imagej.net/distros/cross-platform/ij${version}.zip";
+    sha256 = "sha256-MGuUdUDuW3s/yGC68rHr6xxzmYScUjdXRawDpc1UQqw=";
+  };
+  nativeBuildInputs = [ copyDesktopItems makeWrapper unzip ];
+  desktopItems = lib.optionalAttrs stdenv.isLinux [
+    desktopItem
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/imagej/default.nix:33]:
```suggestion
    (makeDesktopItem {
      name = "ImageJ";
      desktopName = "ImageJ";
      icon = "imagej";
      categories = "Science;Utility;Graphics;";
      exec = "imagej";
    })
```

**Pull request merged**